src/sidebar/store/modules/annotations.ts

```diff
@@ -80,73 +111,44 @@ function findByTag(annotations, tag) {
  * `annotation` may either be new (unsaved) or a persisted annotation retrieved
  * from the service.
  *
- * @param {Omit<Annotation, '$anchorTimeout'>} annotation
- * @param {string} tag - The `$tag` value that should be used for this if it
- *                       doesn't have a `$tag` already
- * @param {string|null} currentUserId - The account id of the currently-auth'd
- *                       user, if any
- * @return {Annotation} - API annotation data with client annotation data merged
+ * @param tag - The `$tag` value that should be used for this if it doesn't have
+ * a `$tag` already
+ * @return - API annotation data with client annotation data merged
  */
-function initializeAnnotation(annotation, tag, currentUserId) {
+function initializeAnnotation(
+  annotation: Omit<Annotation, '$anchorTimeout'>,
+  tag: string,
+  currentUserId: string | null
+): Annotation {
   let orphan = annotation.$orphan;
 
   if (!annotation.id) {
-    // New annotations must be anchored
+    // Unsaved (new) annotations must be anchored
     orphan = false;
   }
 
-  let $cluster = /** @type {HighlightCluster} */ ('other-content');
+  let $cluster: HighlightCluster = 'other-content';
   if (annotation.user === currentUserId) {
     $cluster = isHighlight(annotation) ? 'user-highlights' : 'user-annotations';
   }
 
-  return Object.assign(
-    {},
-    annotation,
-    /** @type {ClientAnnotationData} */ ({
-      $anchorTimeout: false,
-      $cluster,
-      $tag: annotation.$tag || tag,
-      $orphan: orphan,
-    })
-  );
+  return Object.assign({}, annotation, {
```

Comment:
This could also be done with `{ ...annotation, field: value, ... }` syntax in new code.

Reply:
Yeah, I'll update this.

I tried to bite my tongue a little in a few spots in this code and make as few actual changes as possible but this seems innocuous.

I can see some follow-up work, is what I'm sayin', overall.